src/js/angular/guides/tour-lib-services/shepherd.service.js

```diff
@@ -0,0 +1,543 @@
+     * @private
+     */
+    this._updateLocalStorage = () => {
+        const activeTour = Shepherd.activeTour;
+        if (!activeTour) {
+            return;
+        }
+        const currentStep = activeTour.getCurrentStep();
+        LocalStorageAdapter.set(GUIDE_NAME, currentStep.tour.options.name);
+        LocalStorageAdapter.set(GUIDE_CURRENT_STEP_ID, currentStep.options.id);
+    }
+
+    /**
+     * Remove all related data to a guid from local store.
+     * @private
+     */
+    this._clearLocaleStorage = () => {
+        LocalStorageAdapter.remove(GUIDE_NAME);
+        LocalStorageAdapter.remove(GUIDE_PAUSE);
+        LocalStorageAdapter.remove(GUIDE_CURRENT_STEP_ID);
+    };
+
+    /**
+     * Creates a Shepherd Tour step and fill it with base properties from <code>stepDescription</code>.
+     * @param $translate
+     * @param stepDescription - step description. The format of a description is:
+     * @param onShow - function which will be executed when popup dialog shown.
+     * @returns created step.
+     * @private
+     */
+    this._toBaseGuideStep = ($translate, stepDescription, onShow) => {
+        return {
+            id: stepDescription.id,
+            title: $translate.instant(stepDescription.title, stepDescription),
+            text: $translate.instant(stepDescription.content, stepDescription),
+            url: stepDescription.url,
+            maxWaitTime: stepDescription.maxWaitTime,
+            cancelIcon: {enabled: true},
+            attachTo: {
+                element: stepDescription.elementSelector,
+                on: stepDescription.placement
+            },
+            modalOverlayOpeningPadding: 0,
+            modalOverlayOpeningRadius: 0,
+            canBePaused: stepDescription.canBePaused,
+            classes: 'guide-dialog',
+            when: {
+                show: () => {
+                    onShow();
+                }
+            }
+        };
+    }
+
+    /**
+     * Created a base button.
+     * @param text - the text of button.
+     * @param isSecondary - if true will be styled as secondary button.
+     * @param action - to be executed when button is clicked.
+     * @returns created button.
+     */
+    this._getButton = (text, action, isSecondary) => {
+        const button = {
+            text: text,
+            classes: isSecondary ? 'btn-lg btn-secondary' : 'btn-lg btn-primary'
+        };
+
+        button.action = () => {
+            action();
+        }
+
+        return button;
+    }
+
+    /**
+     * Fetches index of step with id <code>stepId</code>.
+     * @param guide - the guide with steps.
+     * @param stepId - the id of step.
+     * @returns {number} - the index of step with id <code>stepId</code> from guide steps array.
+     * @private
+     */
+    this._getStepIndex = (guide, stepId) => {
+        const stepIndex = guide.steps.findIndex((value => value.options.id === stepId));
+        return stepIndex > -1 ? stepIndex : 0;
```

Comment:
Is it OK to return `0` if the step is not found? Assuming that `0` is the index of the first step, if any at all, you would return an index which is not what the caller might expect.

Reply:
I removed function, it was useless.